test/integ/pandas/test_arrow_pandas.py

```diff
@@ -908,3 +909,45 @@ def test_query_resultscan_combos(conn_cnx, query_format, resultscan_format):
         if isinstance(scanned_results, pandas.DataFrame):
             scanned_results = [tuple(e) for e in scanned_results.values.tolist()]
         assert results == scanned_results
+
+
+@pytest.mark.parametrize(
```

Comment:
Alternative:
Parameterize `use_decimal` and `expected_type` together instead of an if-else case with two asserts.
e.g.
```
@pytest.mark.parametrize("use_decimal,expected", [(False, numpy.float64), (True, decimal.Decimal)])
def test_number_fetchall_retrieve_type(conn_cnx, use_decimal, expected):
```

Reply:
Good idea! I'll make this change 👍